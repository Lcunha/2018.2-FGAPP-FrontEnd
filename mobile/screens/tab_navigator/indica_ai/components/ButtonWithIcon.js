import React, { Component } from 'react'
import {
    View,
    StyleSheet
} from 'react-native'
import {
    Text,
    Button,
    Icon
} from 'native-base'

class ButtonWithIcon extends Component {

    constructor(props) {
        super(props)
        this.state = {
            label: props.label,
            icon: props.icon,
            onPress: props.onPress
        }
    }

    render () {

        const {
            label,
            icon,
            onPress,
        } = this.state

        return (
          <Button
            style={styles.registerButton}
            block
            info
            iconLeft
            onPress={onPress}
          >
            <Icon
              name={icon}
              color='white'
              size={25}
            />
            <Text style={{color: 'white'}}>{label}</Text>
          </Button>
        )
    }
}

export default ButtonWithIcon

const styles = StyleSheet.create({
  registerButton: {
    top: 30,
    marginHorizontal: 10
  }

});
