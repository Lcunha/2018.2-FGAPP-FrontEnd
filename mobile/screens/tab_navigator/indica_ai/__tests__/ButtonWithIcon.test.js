import React from 'react';
import ButtonWithIcon from '../components/ButtonWithIcon';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';

Enzyme.configure({adapter: new Adapter()});

  it('renders correctly', () => {
    const tree = renderer.create(<ButtonWithIcon />).toJSON();
    expect(tree).toMatchSnapshot();
});
