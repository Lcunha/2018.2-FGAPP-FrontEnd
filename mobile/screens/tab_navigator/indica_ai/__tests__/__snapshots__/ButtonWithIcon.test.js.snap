// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#62B1F6",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "borderWidth": undefined,
      "elevation": 2,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "center",
      "marginHorizontal": 10,
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": undefined,
      "shadowOffset": undefined,
      "shadowOpacity": undefined,
      "shadowRadius": undefined,
      "top": 30,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        },
        Object {
          "color": "white",
        },
      ]
    }
    uppercase={false}
    virtual={undefined}
  />
</View>
`;
